compiler-rt: un-unset COMPILER_RT_BUILD_CRT

https://reviews.llvm.org/D155126 added this `unset` as a workaround for bots.
Carrying it means we build CRT bits for baremetal targets, which isn't what we
want to do.

diff --git a/compiler-rt/lib/builtins/CMakeLists.txt b/compiler-rt/lib/builtins/CMakeLists.txt
index 7f00391d1dce..1f449063e06b 100644
--- a/compiler-rt/lib/builtins/CMakeLists.txt
+++ b/compiler-rt/lib/builtins/CMakeLists.txt
@@ -898,7 +898,6 @@ endif()
 # TODO: COMPILER_RT_BUILD_CRT used to be a cached variable so we need to unset
 # it first so cmake_dependent_option can set the local variable of the same
 # name. This statement can be removed in the future.
-unset(COMPILER_RT_BUILD_CRT CACHE)
 
 cmake_dependent_option(COMPILER_RT_BUILD_CRT "Build crtbegin.o/crtend.o" ON "COMPILER_RT_HAS_CRT" OFF)
 
